---
title: Processing the kong2012_healthy dataset
author: Yi-Wen Hsiao
date: "`r format(Sys.Date(), '%d %B %Y')`"
output:
  BiocStyle::html_document:
    titlecaps: false
    toc_float: true
bibliography: ref.bib
---

```{r style, echo=FALSE, results='hide', message=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
```

# Downloading the expression data and metadata data

We obtain a bulk RNA sequencing dataset of the healthy subtypes from @kong2012characteristics.
Expression matrix for endogenous genes are available from the Gene Expression Omnibus
using the accession number [GSE18123](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18123).
We download them using the `r Biocpkg("GEOquery")` package.

```{r}
# import libraries
library(GEOquery)
library(biomaRt)
library(stringr)
library(dplyr)

# fetch data based on 
gse_no <-'GSE18123'
gse <- getGEO(gse_no, GSEMatrix=TRUE, destdir=tempdir())
gse
```

We preclean the metadata.

```{r}
# GPL570
metadata_gpl570 <- pData(phenoData(gse[[1]]))[,c(1,6,8)]
metadata_gpl570 <- metadata_gpl570[str_detect(metadata_gpl570$title, 'M-'),]
metadata_gpl570$title <- 'healthy'
metadata_gpl570$gsm <- row.names(metadata_gpl570)
metadata_gpl570$sample_id <- paste0(gse_no, '_', row.names(metadata_gpl570), '_',metadata_gpl570$title)
metadata_gpl570 <- metadata_gpl570[,c('title', 'sample_id' ,'gsm')]
names(metadata_gpl570)[1] <- 'subtype'
metadata_gpl570$study_id <- gse_no
metadata_gpl570$platform <- 'HG-U133_Plus_2'
head(metadata_gpl570)

# GPL6244
metadata_gpl6244 <- pData(phenoData(gse[[2]]))[,c(1,6,8)]
metadata_tmp1 <- metadata_gpl6244[str_detect(metadata_gpl6244$title, 'M-'),]
metadata_tmp2 <- metadata_gpl6244[str_detect(metadata_gpl6244$title, 'F-'),]
metadata_gpl6244 <- rbind(metadata_tmp1,metadata_tmp2)
metadata_gpl6244$title <- 'healthy'
metadata_gpl6244$gsm <- row.names(metadata_gpl6244)
metadata_gpl6244$sample_id <- paste0(gse_no, '_', row.names(metadata_gpl6244), '_',metadata_gpl6244$title)
metadata_gpl6244 <- metadata_gpl6244[,c('title', 'sample_id' ,'gsm')]
names(metadata_gpl6244)[1] <- 'subtype'
metadata_gpl6244$study_id <- gse_no
metadata_gpl6244$platform <- 'HuGene-1_0-st'
head(metadata_gpl6244)

# merge them together
metadata <- rbind(metadata_gpl570, metadata_gpl6244)
head(metadata)
```


We also preprocess the expression matrix.

```{r}
# GPL570
exp <- exprs(gse[[1]])
gpl <- getGEO('GPL570', destdir=tempdir())
colnames(Table(gpl))
annot <- Table(gpl)[,c("ID","Gene Symbol")]
names(annot) <- c('probe_id', 'gene_symbol')
annot<- annot[!(annot$gene_symbol == ''),]
annot<- annot[!grepl("orf", annot$gene_symbol),]
annot<- annot[!grepl("ORF", annot$gene_symbol),]
annot<- annot[!grepl("LINC", annot$gene_symbol),]
annot<- annot[!grepl("MIR", annot$gene_symbol),]
annot<- annot[!grepl("LOC", annot$gene_symbol),]
annot<- annot[!grepl("\\/\\/", annot$gene_symbol),]
exp_annot <- merge(annot, exp, by.x = 'probe_id', by.y = 'row.names')[,-1]
exp_annot_final <- exp_annot %>%
  group_by(gene_symbol) %>%
  summarise_all("mean") %>% as.data.frame()
row.names(exp_annot_final) <- exp_annot_final$gene_symbol
exp_annot_final <- exp_annot_final[,-1]
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
sym2ens <- getBM(attributes=c('external_gene_name','ensembl_gene_id'),
      filters = 'external_gene_name',
      values = row.names(exp_annot_final),
      mart = ensembl)
names(sym2ens) <- c('gene_symbol', 'ensembl_id')
gene_annot <- merge(annot, sym2ens, by = 'gene_symbol')
gene_annot <- unique(gene_annot[,-2])
exp_annot_ensembl <- merge(gene_annot, exp_annot_final, by.x = 'gene_symbol', by.y = 'row.names')
row.names(exp_annot_ensembl) <- exp_annot_ensembl$ensembl_id
exp_annot_ensembl_gpl570 <- exp_annot_ensembl[,-1:-2]
head(exp_annot_ensembl_gpl570[,1:5])

# GPL6244
exp <- exprs(gse[[2]])
gpl <- getGEO('GPL6244', destdir=tempdir())
colnames(Table(gpl))
annot <- Table(gpl)[,c("ID","gene_assignment")]
annot <- annot[-1,]
to_clean <- str_split(annot$gene_assignment,' // ')
for (i in 1:length(to_clean)){
  #i = 1
  annot$gene_assignment[i] <- to_clean[[i]][2]
}
names(annot) <- c('probe_id', 'gene_symbol')
annot<- annot[!(is.na(annot$gene_symbol)),]
annot<- annot[!(annot$gene_symbol == ''),]
annot<- annot[!grepl("orf", annot$gene_symbol),]
annot<- annot[!grepl("ORF", annot$gene_symbol),]
annot<- annot[!grepl("LINC", annot$gene_symbol),]
annot<- annot[!grepl("MIR", annot$gene_symbol),]
annot<- annot[!grepl("LOC", annot$gene_symbol),]
annot
exp_annot <- merge(annot, exp, by.x = 'probe_id', by.y = 'row.names')[,-1]
exp_annot_final <- exp_annot %>%
  group_by(gene_symbol) %>%
  summarise_all("mean") %>% as.data.frame()
row.names(exp_annot_final) <- exp_annot_final$gene_symbol
exp_annot_final <- exp_annot_final[,-1]
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
sym2ens <- getBM(attributes=c('external_gene_name','ensembl_gene_id'),
      filters = 'external_gene_name',
      values = row.names(exp_annot_final),
      mart = ensembl)
names(sym2ens) <- c('gene_symbol', 'ensembl_id')
gene_annot <- merge(annot, sym2ens, by = 'gene_symbol')
gene_annot <- unique(gene_annot[,-2])
exp_annot_ensembl <- merge(gene_annot, exp_annot_final, by.x = 'gene_symbol', by.y = 'row.names')
row.names(exp_annot_ensembl) <- exp_annot_ensembl$ensembl_id
exp_annot_ensembl_gpl6244 <- exp_annot_ensembl[,-1:-2]
head(exp_annot_ensembl_gpl6244[,1:5])

# merge them together
exp_annot_ensembl <- merge(exp_annot_ensembl_gpl570,exp_annot_ensembl_gpl6244, by = 'row.names')

```

We check that the sample are matched in the same order in both expression matrix and metadata.

```{r}
# sample selection
common_smp <- intersect(names(exp_annot_ensembl), metadata$gsm)
exp_subset <- exp_annot_ensembl[,common_smp]
metadata <- metadata[common_smp,]
names(exp_subset) <- metadata$sample_id
metadata <- metadata[,names(metadata)!= 'gsm']
row.names(metadata) <- metadata$sample_id
metadata <- metadata[,c('study_id','sample_id', 'subtype', 'platform')]
all(row.names(metadata)==names(exp_subset))
```

We transform the expression value into rank scale.
```{r}
rankTrans <- function(exp_data){
  transformed_data <- (apply(exp_data, 2, rank)-1)/(nrow(exp_data)-1)
  transformed_data
}
rankdata <- rankTrans(exp_subset)
```


# Saving to file 

We now save all of the components to file for upload to `r Biocpkg("ExperimentHub")`.
These will be used to construct a `DataFrame` on the client side when the dataset is requested.

```{r}

path <- file.path("leukemiaAtlas", "kong2012_healthy")
dir.create(path, showWarnings=FALSE, recursive=TRUE)
saveRDS(rankdata, file=file.path(path, "rankData.rds"))
saveRDS(metadata , file=file.path(path, "colData.rds"))
```

# Session information

```{r}
sessionInfo()
```

# References
