---
title: Processing the kohlmann2008_all dataset
author: Yi-Wen Hsiao
date: "`r format(Sys.Date(), '%d %B %Y')`"
output:
  BiocStyle::html_document:
    titlecaps: false
    toc_float: true
bibliography: ref.bib
---

```{r style, echo=FALSE, results='hide', message=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
```

# Downloading the expression data and metadata data

We obtain a bulk RNA sequencing dataset of the AML subtypes from @kohlmann2008international.
Expression matrix for endogenous genes are available from the Gene Expression Omnibus
using the accession number [GSE13159](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13159).
We download them using the `r Biocpkg("BiocFileCache")` package.

```{r}
# import libraries

library(BiocFileCache)
library(GEOquery)
library(biomaRt)
library(stringr)
library(dplyr)

# fetch data
bfc <- BiocFileCache("raw_data", ask = FALSE)
base.url <- file.path("ftp://ftp.ncbi.nlm.nih.gov/geo",
    "series/GSE13nnn/GSE13159/matrix")
exp.file <- bfcrpath(bfc, file.path(base.url, 
    "GSE13159_series_matrix.txt.gz"))
```

We preclean the metadata.

```{r}
metadata <- as.data.frame(data.table::fread(file=exp.file, nrows = 67, fill=TRUE))
metadata <- metadata[metadata[,1]%in%c('!Sample_characteristics_ch1','!Sample_geo_accession'),]
metadata <- as.data.frame(t(metadata[c(1,3),-1]))
names(metadata) <- c('gsm','subtype')
metadata$subtype[str_detect(metadata$subtype, 'ALL')] <- 'ALL'
metadata$subtype[str_detect(metadata$subtype, 'AML')] <- 'AML'
metadata$subtype[str_detect(metadata$subtype, 'CLL')] <- 'CLL'
metadata$subtype[str_detect(metadata$subtype, 'MDS')] <- 'MDS'
metadata$subtype[str_detect(metadata$subtype, 'CML')] <- 'CML'
metadata$subtype[str_detect(metadata$subtype, 'Non')] <- 'healthy'
metadata <- metadata[metadata$subtype!='MDS',]
metadata$sample_id <- paste0('GSE13159_', metadata$gsm, '_', metadata$subtype)
metadata$study_id <- 'GSE13159'
metadata$platform <- 'HG-U133_Plus_2'
row.names(metadata) <- metadata$gsm
head(metadata)
```


We also preprocess the expression matrix.

```{r}
# expression matrix
exp <- as.data.frame(data.table::fread(exp.file, skip = 67))
row.names(exp) <- exp[,1]
exp <- exp[,-1]
# clean probe information (probe2symbol)
gpl <- getGEO('GPL570', destdir=tempdir())
colnames(Table(gpl))
annot <- Table(gpl)[,c("ID","Gene Symbol")]
names(annot) <- c('probe_id', 'gene_symbol')
annot<- annot[!(annot$gene_symbol == ''),]
annot<- annot[!grepl("orf", annot$gene_symbol),]
annot<- annot[!grepl("ORF", annot$gene_symbol),]
annot<- annot[!grepl("LINC", annot$gene_symbol),]
annot<- annot[!grepl("MIR", annot$gene_symbol),]
annot<- annot[!grepl("LOC", annot$gene_symbol),]
annot<- annot[!grepl("\\/\\/", annot$gene_symbol),]
exp_annot <- merge(annot, exp, by.x = 'probe_id', by.y = 'row.names')[,-1]
exp_annot_final <- exp_annot %>%
  group_by(gene_symbol) %>%
  summarise_all("mean") %>% as.data.frame()
row.names(exp_annot_final) <- exp_annot_final$gene_symbol
exp_annot_final <- exp_annot_final[,-1]

# symbol2ensembl
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
sym2ens <- getBM(attributes=c('external_gene_name','ensembl_gene_id'),
      filters = 'external_gene_name',
      values = row.names(exp_annot_final),
      mart = ensembl)
names(sym2ens) <- c('gene_symbol', 'ensembl_id')
gene_annot <- merge(annot, sym2ens, by = 'gene_symbol')
gene_annot <- unique(gene_annot[,-2])
exp_annot_ensembl <- merge(gene_annot, exp_annot_final, by.x = 'gene_symbol', by.y = 'row.names')
row.names(exp_annot_ensembl) <- exp_annot_ensembl$ensembl_id
exp_annot_ensembl <- exp_annot_ensembl[,-1:-2]
head(exp_annot_ensembl[,1:5])
```

We check that the sample are matched in the same order in both expression matrix and metadata.

```{r}
# sample selection
common_smp <- intersect(names(exp_annot_ensembl), metadata$gsm)
exp_subset <- exp_annot_ensembl[,common_smp]
metadata <- metadata[common_smp,]
names(exp_subset) <- metadata$sample_id
metadata <- metadata[,names(metadata)!= 'gsm']
row.names(metadata) <- metadata$sample_id
metadata <- metadata[,c('study_id','sample_id', 'subtype', 'platform')]
all(row.names(metadata)==names(exp_subset))
```

We transform the expression value into rank scale.
```{r}
rankTrans <- function(exp_data){
  transformed_data <- (apply(exp_data, 2, rank)-1)/(nrow(exp_data)-1)
  transformed_data
}
rankdata <- rankTrans(exp_subset)
dim(rankdata)
summary(rankdata[,1:5])
```


# Saving to file 

We now save all of the components to file for upload to `r Biocpkg("ExperimentHub")`.
These will be used to construct a `DataFrame` on the client side when the dataset is requested.

```{r}

path <- file.path("leukemiaAtlas", "kohlmann2008_all")
dir.create(path, showWarnings=FALSE, recursive=TRUE)
saveRDS(rankdata, file=file.path(path, "rankData.rds"))
saveRDS(metadata, file=file.path(path, "colData.rds"))
```

# Session information

```{r}
sessionInfo()
```

# References
